### PR TITLE
fix: metadata is overwritten for child managed eventType when updating parent

### PR DESCRIPTION
**Pull request by anikdhabal**
## What does this PR do?

- Fixes #XXXX (GitHub issue number)
- Fixes CAL-XXXX (Linear issue number - should be visible at the bottom of the GitHub issue description)

More Context:- https://app.campsite.com/cal/posts/z7n1o8dexajx#comment-wvo37qrhchc6

### CODE REVIEW

File: packages/features/ee/managed-event-types/lib/handleChildrenEventTypes.ts [hunk 1]
```diff
@@ -249,6 +264,13 @@ export default async function handleChildrenEventTypes({
                 : {
                     deleteMany: {},
                   },
+            metadata: {
+              ...(eventType.metadata as Prisma.JsonObject),
+              ...(metadata?.multipleDuration && "length" in unlockedFieldProps
+                ? { multipleDuration: metadata.multipleDuration }
+                : {}),
+              ...(metadata?.apps && "apps" in unlockedFieldProps ? { apps: metadata.apps } : {}),
+            },
```
Comment (anikdhabal) [inline packages/features/ee/managed-event-types/lib/handleChildrenEventTypes.ts:273]: Currently, if any fields are unlocked for a child eventType and specific values are set, they are replaced by the parent eventType's metadata values when updating.

**Review by Udit-takkar — DISMISSED** [review]
unit tests are failing

File: packages/features/ee/managed-event-types/lib/handleChildrenEventTypes.ts (hunk 1 above)
Comment (Udit-takkar) [reply packages/features/ee/managed-event-types/lib/handleChildrenEventTypes.ts:265]: why don't you do the same for other fields?

Comment (anikdhabal) [reply packages/features/ee/managed-event-types/lib/handleChildrenEventTypes.ts:266]: All the fields aren't saved inside the metadata and are not overwritten. These two fields are the most common and high priority and shouldn't be overwritten

**Review by Udit-takkar — APPROVED** [review]
LGTM

**Pull request merged**